website/registrations/utils.py

```diff
@@ -0,0 +1,349 @@
+# -*- coding: utf-8 -*-
+import re
+import csv
+import io
+import functools
+import json
+
+from rest_framework.exceptions import NotFound, ValidationError
+
+from osf.models.licenses import NodeLicense
+from osf.models import AbstractNode, RegistrationProvider, Subject, RegistrationSchema, Institution
+from website import settings
+
+
+METADATA_FIELDS = {'title': {'format': 'string', 'required': True},
+                   'description': {'format': 'string', 'required': True},
+                   'admin': {'format': 'list', 'required': True},
+                   'read-write': {'format': 'list'},
+                   'read-only': {'format': 'list'},
+                   'bibliographic contributors': {'format': 'list'},
+                   'category': {'format': 'string'},
+                   'affiliated institutions': {'format': 'list'},
+                   'license': {'format': 'object', 'required': True},
+                   'subjects': {'format': 'list', 'required': True},
+                   'tags': {'format': 'list'},
+                   'project guid': {'format': 'string'},
+                   'external id': {'format': 'string'}}
+CONTRIBUTOR_METADATA_FIELDS = ['admin', 'read-write', 'read-only', 'bibliographic contributors']
+
+class BulkRegistrationUpload():
+    @property
+    def is_valid(self):
+        return not len(self.errors)
+
+    def __init__(self, bulk_upload_csv, provider_id):
+        self.raw_csv = bulk_upload_csv.read()
+        # self.raw_csv = bulk_upload_csv.read().decode('utf-8')
+        self.reader = csv.DictReader(io.StringIO(self.raw_csv))
+        self.headers = [header.lower() for header in self.reader.fieldnames]
+        schema_id_row = next(self.reader)
+        self.schema_id = schema_id_row[self.reader.fieldnames[0]]
+        self.provider_id = provider_id
+        if self.schema_id:
+            try:
+                self.registration_provider = RegistrationProvider.load(self.provider_id)
+                self.registration_schema = self.registration_provider.schemas.get(_id=self.schema_id)
+            except RegistrationSchema.DoesNotExist:
+                raise NotFound(detail='Schema with id "{}" was not found'.format(self.schema_id))
+            except RegistrationProvider.DoesNotExist:
+                raise NotFound(detail='Registration provider with id "{}" was not found').format(self.provider_id)
+        self.schema_questions = BulkRegistrationUpload.get_questions_validations(self.registration_schema)
```

Comment:
It seems we're getting the validations "objects" (dict) for the schema questions here. So maybe call it `schema_questions_validations` or something similar?